nidx/src/maintenance/scheduler.rs

```diff
@@ -214,6 +330,148 @@ mod tests {
 
         use crate::metadata::{Index, IndexId, IndexKind, NidxMetadata, Shard};
 
+        fn merge_scheduler() -> MergeScheduler {
+            MergeScheduler::from_settings(&MergeSettings {
+                min_number_of_segments: 3,
+                ..Default::default()
+            })
+        }
+
+        async fn create_segments(pool: &sqlx::PgPool, index: &Index, segment_records: &[i64]) -> anyhow::Result<Seq> {
```

Comment:
nit: I would like if this function name explained why it's receiving a vec! of records, something like `create_segments_with_num_records`